build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -474,13 +474,17 @@ public void nextNodeToNextVersion() {
         commonNodeConfig();
         nodeIndex += 1;
         if (node.getTestDistribution().equals(TestDistribution.DEFAULT)) {
-            if (node.getVersion().onOrAfter("7.16.0")) {
+            if (hasDeprecationIndexing(node)) {
                 node.setting("cluster.deprecation_indexing.enabled", "false");
             }
         }
         node.start();
     }
 
+    private static boolean hasDeprecationIndexing(ElasticsearchNode node) {
+        return node.getVersion().onOrAfter("7.16.0") && node.getSettingKeys().contains("stateless.enabled") == false;
```

Comment:
I'll admit that this is a bit of a hack, and I'm open to other suggestions.
The constraints that I ran into are:
- `elasticsearch.serverless-run.gradle.kts` configures a test cluster directly (not a spec) so it can't use a settings provider (which is how we solved this elsewhere)
- `ElasticsearchNode` does not expose which modules exist, so I cannot make this dependent on the deprecation module existing
- Serverless doesn't appear to be an exposed concept (as a distribution type, or version number, etc) in `ElasticsearchNode` (which is generally a good thing, but left me with very few options here).

This change allows `./gradlew run` to work in serverless again, which is a good thing, and I would like to make that happen. I am, however, very happy to iterate on this in order to come up with a cleaner solution if people don't like this one.

Reply:
This is fine by me. I think the long term solution here is to have `gradlew run` use the new test clusters infrastructure rather than `ElasticsearchNode` since it's considerably more extensible.